### PR TITLE
Solved Validation errors

### PR DESCRIPTION
**Pull request by RidhamShah**
Scheduler - Solve validation error for changing the state of the Experiment to `Scheduled`

Segments - when importing an old experiment, it doesn't have segments (but gets created). So on updating anything, it throws a validation error of missing `Segment` data. So to solve it, while importing it will create segment data for that experiment.
So errors can be solved by deleting and re-importing experiments.

Condition Payload - Solve validation errors of `ConditionPayload` while updating an Experiment.

### CODE REVIEW

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts [hunk 1]
```diff
@@ -485,6 +511,7 @@ export class ImportExperimentComponent implements OnInit {
     result = this.deduceConditionPayload(result);
     result = this.deducePartition(result);
     result = this.deduceFactors(result);
+    result = this.deduceParticipants(result);
```
Comment (VivekFitkariwala) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts:514]: result itself is getting mutated inside the deduce*** functions. We don't need to return anything from the function.

**Review by VivekFitkariwala — CHANGES_REQUESTED** [review]
One comment need to be addressed.

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts (hunk 1 above)
Comment (RidhamShah) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts]: code updated

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts [hunk 2]
```diff
@@ -511,7 +509,7 @@ export class ImportExperimentComponent implements OnInit {
     result = this.deduceConditionPayload(result);
```
Comment (VivekFitkariwala) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts:509]: All the deduce function is mutating the result object. So you don't need to return anything.

**Review by VivekFitkariwala — APPROVED** [review]
One more comment to address.

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts (hunk 2 above)
Comment (RidhamShah) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts]: code updated

**Comment by zackcl** [comment]
I can confirm the issue https://github.com/CarnegieLearningWeb/UpGrade/issues/1074 has been fixed. Is this PR ready to be merged?

**Comment by zackcl** [comment]
@RidhamShah As I mentioned in the meeting, if I create and export an experiment with custom payloads in the `dev` branch, and try to import the experiment in this branch, I get the following error in the backend and not able to import the experiment.

```
QueryFailedError: Error in creating conditions, decision points, conditionPayloads and queries "addExperimentInDB"
    at QueryFailedError.TypeORMError [as constructor] (/Users/zlee/Desktop/code/UpGrade/backend/packages/Upgrade/src/error/TypeORMError.ts:7:9)
    at new QueryFailedError (/Users/zlee/Desktop/code/UpGrade/backend/packages/Upgrade/src/error/QueryFailedError.ts:9:9)
    at PostgresQueryRunner.<anonymous> (/Users/zlee/Desktop/code/UpGrade/backend/packages/Upgrade/src/driver/postgres/PostgresQueryRunner.ts:258:19)
    at step (/Users/zlee/Desktop/code/UpGrade/backend/packages/Upgrade/node_modules/tslib/tslib.js:143:27)
    at Object.throw (/Users/zlee/Desktop/code/UpGrade/backend/packages/Upgrade/node_modules/tslib/tslib.js:124:57)
    at rejected (/Users/zlee/Desktop/code/UpGrade/backend/packages/Upgrade/node_modules/tslib/tslib.js:115:69)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

But if I import the experiment that was created and exported in this branch, I don't get this error and it works fine. So it seems like there's a difference between the exported JSON file in the `dev` branch and this branch.

Here's a sample experiment file exported from the `dev` branch. You will get an error if you try to import this file in this branch.

[fewfwefwe.json](https://github.com/CarnegieLearningWeb/UpGrade/files/13192939/fewfwefwe.json)

**Comment by zackcl** [comment]
I was able to properly import the experiment JSON file without the issue after clearing the Postgres DB.

**Review by zackcl — APPROVED** [review]

**Pull request merged**